crates/net/network/src/transactions.rs

```diff
@@ -475,29 +475,37 @@ where
         }
     }
 
-    fn report_bad_message(&self, peer_id: PeerId, req_err: Option<RequestError>) {
-        let msg = match req_err {
-            Some(RequestError::UnsupportedCapability) => {
-                self.network.reputation_change(peer_id, ReputationChangeKind::BadProtocol);
-                "Penalizing peer for unsupported capability"
-            }
-            Some(RequestError::Timeout) => {
-                self.network.reputation_change(peer_id, ReputationChangeKind::Timeout);
-                "Penalizing peer for timeout"
-            }
-            Some(RequestError::ChannelClosed | RequestError::ConnectionDropped) => {
-                self.network.reputation_change(peer_id, ReputationChangeKind::Dropped);
-                "Penalizing peer for dropped connection"
-            }
-            Some(RequestError::BadResponse) | None => {
-                self.network.reputation_change(peer_id, ReputationChangeKind::BadTransactions);
-                "Penalizing peer for bad transaction"
+    fn report_message(&self, peer_id: PeerId, kind: ReputationChangeKind) {
+        let msg = match kind {
+            ReputationChangeKind::BadProtocol => "Penalizing peer for bad protocol rule",
+            ReputationChangeKind::Timeout => "Penalizing peer for timeout",
+            ReputationChangeKind::Dropped => "Penalizing peer for dropped connection",
+            ReputationChangeKind::BadTransactions => "Penalizing peer for bad transaction",
+            ReputationChangeKind::AlreadySeenTransaction => {
+                "Penalizing peer for already seen transaction"
             }
+            ReputationChangeKind::BadBlock => "Penalizing peer for bad block",
+            ReputationChangeKind::BadMessage => "Penalizing peer for bad message",
```

Comment:
I think using the debug impl for this is sufficient, so no need to derive a message